### PR TITLE
ThumbnailStore fixes

### PR DESCRIPTION
**Pull request by atarkowska**
# What this PR does

additional commits from https://github.com/openmicroscopy/openmicroscopy/pull/5158

revert https://github.com/openmicroscopy/openmicroscopy/pull/5173

# Testing this PR

test should pass now
https://idr-ci.openmicroscopy.org:8443/job/OMERO-test-integration/lastCompletedBuild/testReport/OmeroPy.test.integration.gatewaytest.test_image/TestImage/testThumbnailSet/

### CODE REVIEW

**Comment by atarkowska** [comment]
Test is green https://idr-ci.openmicroscopy.org:8443/job/OMERO-test-integration/lastCompletedBuild/testReport/OmeroPy.test.integration.gatewaytest.test_image/

**Comment by joshmoore** [comment]
I wonder if lines 975-978:

```
                if (inProgress && !PROGRESS_VERSION.equals(thumbnailMetadata.getVersion())) {
                    thumbnailMetadata.setVersion(PROGRESS_VERSION);
                    dirtyMetadata = true;
                }
```

should go *after* line 985 (`byte[] thumbnail = retrieveThumbnail(false);`) since if retrieve thumbnail throws, there's no reason to modify `thumbnailMetadata`.

**Comment by atarkowska** [comment]
@joshmoore pushed, but that didn't fix the failing batch, as the failure is not on `retrieveThumbnail`

**Comment by jburel** [comment]
That will match the implementation of ``retrieveThumbnailAndUpdateMetadata``

**Comment by jburel** [comment]
@aleksandra-tarkowska could port your last commit to develop?

**Comment by aleksandra-tarkowska** [comment]
@joshmoore skip thumbnails works

firefoxscreensnapz050

**Comment by atarkowska** [comment]
@joshmoore skipping dirty thumbnails now works, would you like to comment on https://github.com/openmicroscopy/openmicroscopy/pull/5188/commits/04aff28984e990872e3502513f9c465844c9cc3a

firefoxscreensnapz050

**Comment by joshmoore** [comment]
https://github.com/openmicroscopy/openmicroscopy/pull/5188/commits/04aff28984e990872e3502513f9c465844c9cc3a does worry me, minimally since the exception is being eaten. Do you have a stacktrace _without_ this commit?

File: components/server/src/ome/services/ThumbnailBean.java [hunk 1]
```diff
@@ -1005,7 +1005,15 @@ public void createThumbnailsByLongestSideSet(Integer size,
         // process due to the possible unloaded Pixels. If we do not,
         // Pixels will be unloaded and we will hit
         // IllegalStateException's when checking update events.
-        iUpdate.saveArray(toSave.toArray(new Thumbnail[toSave.size()]));
+        try {
+        	iUpdate.saveArray(toSave.toArray(new Thumbnail[toSave.size()]));
+        } catch (Exception e) {
```
Comment (mtbc) [inline components/server/src/ome/services/ThumbnailBean.java:1010]: can this catch be more specific? also, maybe comment what it means if it happens?

Comment (atarkowska) [reply components/server/src/ome/services/ThumbnailBean.java:1010]: I catch https://github.com/openmicroscopy/openmicroscopy/commit/ccec2d305624c95b446434434538a5c03813d200#diff-bae9d2b989f3f6e6de49048fa4dcd34dR1009

but then was throwing

```
throws serverStackTrace = ome.conditions.InternalException:  Wrapped Exception: (org.hibernate.NonUniqueObjectException):
a different object with the same identifier value was already associated with the session: [ome.model.core.Pixels#692188]
```

so I went for Exception to check if that would work at all

**Comment by joshmoore** [comment]
Testing on test34 has shown no issues (all of rohnB opened; most other plates opened including multiple fields)

Merging for a rebuild of IDR 0.3.4. This may ultimately be reverted but it significantly improves the status of bad-thumbnail handling (from "a whole block of thumbnails is missing" to "only the single bad thumbnail is missing".

**Pull request merged**